### PR TITLE
[TS][SDH] LPS-164769 Avoid NPE when open notification has deleted user or layout.

### PR DESCRIPTION
**Pull request by binhtran92**
Hi team,
I'm supporting the ticket: https://issues.liferay.com/browse/LPP-46699 This is an SDH ticket. 
Please help me to review it.
Thank you.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_metadata/metadata_entry.jsp [hunk 1]
```diff
@@ -87,7 +87,13 @@ else if (metadataField.equals("view-count")) {
 	<c:when test='<%= Objects.equals(value, "author") %>'>
 
 		<%
-		User assetRendererUser = UserLocalServiceUtil.getUser(assetRenderer.getUserId());
+		User assetRendererUser = null;
+
+		try {
+			assetRendererUser = UserLocalServiceUtil.getUser(assetRenderer.getUserId());
```
Comment (AliciaGarciaGarcia) [inline modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_metadata/metadata_entry.jsp:93]: If we will do nothing on the exception.
It not better idea to use the `fetch` methods?

File: modules/apps/portal-workflow/portal-workflow-task-web/src/main/resources/META-INF/resources/edit_workflow_task.jsp [hunk 2]
```diff
@@ -45,7 +51,11 @@ if (assetRenderer != null) {
 	assetRendererFactory = assetRenderer.getAssetRendererFactory();
 
 	if (assetRendererFactory != null) {
-		assetEntry = assetRendererFactory.getAssetEntry(workflowHandler.getClassName(), assetRenderer.getClassPK());
+		try {
+			assetEntry = assetRendererFactory.getAssetEntry(workflowHandler.getClassName(), assetRenderer.getClassPK());
```
Comment (AliciaGarciaGarcia) [inline modules/apps/portal-workflow/portal-workflow-task-web/src/main/resources/META-INF/resources/edit_workflow_task.jsp:55]: `assetRenderer` can be null here as you put before?

And we are trying to access to her values?

Checking null maybe it's needed?

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_metadata/metadata_entry.jsp [hunk 3]
```diff
@@ -87,7 +87,13 @@ else if (metadataField.equals("view-count")) {
...
+		}
+		catch (Exception e) {
+		}
```
[the remark this answers is not in the document]
Comment (binhtran92) [reply modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_metadata/metadata_entry.jsp:96]: @adolfopa : Thank you for reviewing, I will update the change.

**Comment by binhtran92** [comment]
Hi @adolfopa @AliciaGarciaGarcia,
Thank you for reviewing, I updated the change.

**Comment by binhtran92** [comment]
ci:test:sf

File: modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/asset/model/LayoutAssetRendererFactory.java [hunk 4]
```diff
@@ -63,15 +63,19 @@ public AssetEntry getAssetEntry(String className, long classPK)
...
-		assetEntry.setUserName(user.getFullName());
+
+		User user = _userLocalService.fetchUser(layout.getUserId());
+
+		if (user != null) {
+			assetEntry.setCompanyId(user.getCompanyId());
+			assetEntry.setUserId(user.getUserId());
+			assetEntry.setUserName(user.getFullName());
+		}
+
```
Comment (adolfopa) [inline modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/asset/model/LayoutAssetRendererFactory.java:78]: This looks wrong. If the user doesn't exist, the assetEntry will be partially updated. If the user is null, you should do something like this:
```
if (user != null) {
    ...
}
else {
    assetEntry.setCompanyId(layout.getCompanyId());
    assetEntry.setUserId(_userLocalService.getDefaultUserId(layout.getCompanyId());
    ...
}
```

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Comment by sontruongces** [comment]
@adolfopa, Can you please take a look at the fix based on Binh's update? The customer needs the fix to Go-live so we want your approval for the fix so we can delivery it to them ASAP. Thanks!

**Review by adolfopa — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
@binhtran92 Please fix the sf error. Before pushing changes, remember to always run source format in you local branch.

**Comment by sontruongces** [comment]
> @binhtran92 Please fix the sf error. Before pushing changes, remember to always run source format in you local branch.

@adolfopa , Binh is out of the office from today to next Monday. So I'm not sure that he can update it. So please wait for his update when he is back. If there is anything that I can help please just let me know. Thanks!

**Review by AliciaGarciaGarcia — APPROVED** [review]
LGTM > SF failures relevant to this PR

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

**Comment by adolfopa** [comment]
@sontruongces If you could fetch this branch, rebase, run auto sf, add an extra SF commit and push-force it here (or send a new pr if you don't have permissions) we'd be eternally grateful :-) Right now we're working on other branches, so it's a bit of a pain for us.

**Comment by binhtran92** [comment]
ci:test:sf

**Comment by binhtran92** [comment]
ci:test:relevant

**Comment by binhtran92** [comment]
Hi guys, thank you for helping me with this PR. I updated the new SF.

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**